quadsim/robots/robot.py

```diff
@@ -1,44 +1,222 @@
-from typing import Any
-from typing import List
+"""Base class for all robots."""
+import numpy as np
+import pybullet_data
 
-# TODO: move Sensor and Motor to appropriate files
-
-
-class Sensor:
-    pass
-
-
-class Motor:
-    pass
+from quadsim.robots import motor_model
 
 
 class Robot:
     """Robot Base."""
 
-    def __init__(
-        self,
-        pybullet_client: Any,
-        motors: List[Motor],
-        sensors: List[Sensor],
-        on_rack: bool = False,
-    ) -> None:
+    def __init__(self, pybullet_client, config) -> None:
         """Constructs a base robot and resets it to the initial states.
 
         Args:
-
           pybullet_client: The instance of BulletClient to manage different
             simulations.
           motors: A list of motor models.
           sensors: A list of sensor models.
           on_rack: Whether the robot is hanging in mid-air or not. Used for debugging.
         """
+        self._pybullet_client = pybullet_client
+        self.config = config
+        self._setup_simulator()
+        self._load_robot_urdf()
+        self._motor_model = motor_model.MotorModel(self.config.motor)
```

Comment:
My last major comment (maybe the only actual one) is that it's slightly confusing that MotorModel actually contains a collection of motors. I know we spoke about this briefly already, and that you mentioned it comes out of computational efficiency.

I wanted to take an additional second to think through whether there was any alternative structure that doesn't sacrifice much and is even clearer. Any thoughts?

I suppose for some robots, it's very possible that there could be a different MotorModel for each motor (imagine some motors are for heavy loads and have one set of properties, while others operate only in low-torque regimes and so they have different properties). Should there then be something more like: motors: Sequence[MotorModel] or is that too much? Maybe instead we just name the class MotorGroup or similar.

Reply:
I agree with what you said. In addition, the performance gain is not that big.

Let's keep the motor-model idea simple then --- I'll refactor the code so that one `MotorModel` takes care of exactly one motor.